### DISCUSSION TITLE
Integration with popular frameworks

### DISCUSSION THREAD

**Discussion by Real-Gecko** [hyperdiv/hyperdiv#70] [Ideas]
Any ideas of how the project can be integrated with some other popular frameworks like Django or Flask? Would be nice to replace existing frontend mess with `hyperdiv`, leaving all background logic intact.

**Comment by mariusnita** [comment]
Hi, I'm interested in integrating with Flask and Django. Haven't had the opportunity to dig in yet.

**Reply by Real-Gecko** [reply]
The only problem I currently see is that Django runs it's own server like `python manage.py runserver` or `gunicorn wsgi.py`, while Hyperdiv runs it's own, really curious how we can start Hyperdiv process from within django.

**Reply by mariusnita** [reply]
My initial thought was to make asgi/wsgi adapters for Hyperdiv so it runs in the same process as flask/django.

**Reply by Real-Gecko** [reply]
Yeah, for running in production wsgi/asgi is definitely a thing. I guess we can connect something like SQLAlchemy to Hyperdiv project, however there's a lot of existing projects on Django that I guess can benefit from this framework. Backend/Frontend splitting with Hyperdiv on the front and Django on the back I guess makes no sense, as it would be nicer to run everything from single source.

**Comment by thedatadavis** [comment]
You could use Django as an api backend that’s hosted separately. Then the
hyperdiv app would be it frontend getting data from the api. (Similar to
how some people use React with django).

*mChris Davis*
512-817-6951 / bootstrapital.com

**Reply by Real-Gecko** [reply]
Yep, I thought about that, but it bring unnecessary complexity to the entire project.

**Comment by thedatadavis** [comment]
There’s fair. I think you’ll find the current tradeoffs on some of the
batteries included in django for the simplicity of frontend development can
be a net positive with the framework.

It’s basically like working with Flask.

*Chris Davis*
512-817-6951 / bootstrapital.com

**Reply by Real-Gecko** [reply]
Yep, would be great for upcoming projects, because switching existing ones is a pain :D

**Comment by zzgvh** [comment]
I'm experimenting with Django + hyperdiv. My use case is internal tools where I can use existing and new Django models and create live dashboards using hyperdiv for the user interface. As I already was using Django, in a Docker environment (https://github.com/cookiecutter/cookiecutter-django), I added a few lines to the `Dockerfile` and `start` file to get hyperdiv running on port 9000. Then I created a new Django app where all my hyperdiv code lives. Sure, it's a hydra, but seems to work well. I plan on using Caddy in front of it all.

This allows me to use the Django models and the admin backend for data management, and to write the front end dashboards using hyperdiv instead of django views. So no HTML/templates/React. Just hyperdiv!

I think this will allow me to also use the Django i18n toolbox. Haven't looked at that yet, but I think it should work.

Still early days, but I really like working with hyperdiv. For some questions around styling of widgets I've had to dig into the source code, but all-in-all the docs are pretty good! Thanks so much for this project @mariusnita !!!
